### PR TITLE
update ECS v4 client

### PR DESCRIPTION
**Pull request by kangyili**
Update ECS V4 client and types.  This PR is used by https://github.com/DataDog/datadog-agent/pull/21836

### CODE REVIEW

**Review by L3n41c — CHANGES_REQUESTED** [review]
I think there is an issue with the fact that `V4FromCurrentTask(…)` uses the same `globalUtil.initV3orV4` `sync.Once` object as `V3orV4FromCurrentTask(…)`.

If both functions are called, only one of the two will be allowed to setup its retrier.
The other will panic when calling `TriggerRetry`.

File: pkg/util/ecs/metadata/clients.go [hunk 1]
```diff
@@ -116,6 +117,30 @@ func V3orV4FromCurrentTask() (v3or4.Client, error) {
...
+			Strategy:          retry.Backoff,
+			InitialRetryDelay: initialRetryDelay,
+			MaxRetryDelay:     maxRetryDelay,
+		})
+	})
+	if err := globalUtil.initRetryV4.TriggerRetry(); err != nil {
+		log.Debugf("ECS metadata v4 client init error: %s", err)
+		return nil, err
+	}
+	return globalUtil.v3or4, nil
```
[the remark this answers is not in the document]
Comment (kangyili) [reply pkg/util/ecs/metadata/clients.go]: Nice catch. I created a new v4 field
https://github.com/DataDog/datadog-agent/pull/23248/commits/9244446431e3dd4ea773617a249cb4ec8fdf94b9

File: pkg/util/ecs/metadata/v3or4/types.go [hunk 2]
```diff
@@ -7,17 +7,24 @@ package v3or4
...
+	LaunchType              string             `json:"LaunchType,omitempty"` // present only in v4
+	ContainerInstanceTags   map[string]string  `json:"ContainerInstanceTags,omitempty"`
+	TaskTags                map[string]string  `json:"TaskTags,omitempty"`
+	EphemeralStorageMetrics map[string]int64   `json:"EphemeralStorageMetrics,omitempty"`
+	ServiceName             string             `json:"ServiceName,omitempty"`
+	VPCID                   string             `json:"VPCID,omitempty"`
+	PullStartedAt           string             `json:"PullStartedAt,omitempty"`
+	PullStoppedAt           string             `json:"PullStoppedAt,omitempty"`
+	ExecutionStoppedAt      string             `json:"ExecutionStoppedAt,omitempty"`
+	AvailabilityZone        string             `json:"AvailabilityZone,omitempty"`
```
[the remark this answers is not in the document]
Comment (kangyili) [reply pkg/util/ecs/metadata/v3or4/types.go:27]: Yes, we will need all fields as we're going to add a new core check to collect ecs tasks. The check will use workloadmeta. 

- https://github.com/DataDog/datadog-agent/pull/21836
- 
- https://github.com/DataDog/datadog-agent/pull/22060

File: pkg/util/ecs/metadata/clients.go [hunk 3]
```diff
@@ -179,3 +206,12 @@ func initV3orV4() error {
...
 }
+
+func initV4() error {
+	client, err := newClientV4ForCurrentTask()
+	if err == nil {
+		globalUtil.v4 = client
+		return nil
+	}
+	return err
+}
```
Comment (L3n41c) [inline pkg/util/ecs/metadata/clients.go:217]: The `if` may not be necessary.

```suggestion
func initV4() (err error) {
	globalUtil.v4, err = newClientV4ForCurrentTask()
	return err
}
```

File: pkg/util/ecs/metadata/v3or4/client_test.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	task, err := client.GetTaskWithTags(context.Background())
+	require.NoError(t, err)
+
+	require.Equal(t, *expected(), *task)
+}
+
+// expected generates an expected Task from ./testdata/task.json
+func expected() *Task {
+	return &Task{
+		ClusterName:   "arn:aws:ecs:us-east-1:123457279990:cluster/ecs-cluster",
```
Comment (L3n41c) [inline pkg/util/ecs/metadata/v3or4/client_test.go:39]: `expected` may not need to be a function.

```suggestion
	require.Equal(t, expected, task)
}

// expected generates an expected Task from ./testdata/task.json
var expected := &Task{
		ClusterName:   "arn:aws:ecs:us-east-1:123457279990:cluster/ecs-cluster",
```

**Review by L3n41c — APPROVED** [review]
I only left minor simplification suggestions.

**Pull request merged**